edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/CloudEndpoint.cs

```diff
@@ -216,25 +221,30 @@ async Task<ISinkResult<IRoutingMessage>> ProcessClientMessagesBatch(string id, L
                     var cp = cloudProxy.Value;
                     try
                     {
-                            List<IMessage> messages = routingMessages
-                                .Select(r => this.cloudEndpoint.messageConverter.ToMessage(r))
-                                .ToList();
-
-                            if (messages.Count == 1)
-                            {
-                                await cp.SendMessageAsync(messages[0]);
-                            }
-                            else
-                            {
-                                await cp.SendMessageBatchAsync(messages);
-                            }
-
-                            return new SinkResult<IRoutingMessage>(routingMessages);
+                        List<IMessage> messages = routingMessages
+                            .Select(r => this.cloudEndpoint.messageConverter.ToMessage(r))
+                            .ToList();
+
+                        if (messages.Count == 1)
+                        {
+                            await cp.SendMessageAsync(messages[0]);
                         }
-                        catch (Exception ex)
+                        else
                         {
-                            return this.HandleException(ex, id, routingMessages);
+                            await cp.SendMessageBatchAsync(messages);
                         }
+
+                        return new SinkResult<IRoutingMessage>(routingMessages);
+                    }
+                    catch (EdgeHubCloudSDKException ex)
+                    {
+                        await cp.CloseAsync();
```

Comment:
We should not close the CloudProxy here.. it should be done in CLoudProxy itself..